app/channel.go

```diff
@@ -1230,8 +1245,11 @@ func (a *App) GetChannelsForUser(teamId string, userId string, includeDeleted bo
 	return result.Data.(*model.ChannelList), nil
 }
 
-func (a *App) GetAllChannels(page, perPage int, includeDeleted bool) (*model.ChannelListWithTeamData, *model.AppError) {
-	result := <-a.Srv.Store.Channel().GetAllChannels(page*perPage, perPage, includeDeleted)
+func (a *App) GetAllChannels(page, perPage int, opts model.ChannelSearchOpts) (*model.ChannelListWithTeamData, *model.AppError) {
+	if opts.ExcludeDefaultChannels {
+		opts.ExcludeChannelNames = a.DefaultChannelNames()
```

Comment:
What happens if i set the `ExcludeChannelNames` to `my-own-channel` and then add the `ExcludeDefaultChannels` to `true`. I would expect to exclude both, default channels and `my-own-channel`.

Reply:
That's a good idea. I'll update this.